source/faucet/main.d

```diff
@@ -272,6 +276,69 @@ public class Faucet : IFaucet
     {
         return this.state.utxos.storage;
     }
+
+    // Get UTXOs owned by us
+    private UTXO[Hash] getOwnedUTXOs () nothrow @safe
```

Comment:
There is one more optimization that you could make. Instead of calling this function within `sendTransaction()`, you could call it each time a block is externalized. And then store it as a field. It's much more efficient that way.

Reply:
What do you think of adding it to the `State`?
The `State` keeps track of the faucet's UTXO set and the known height.

How do I check within the faucet when a new block is externalized?

Reply:
https://github.com/bpfkorea/faucet/blob/v0.x.x/source/faucet/main.d#L71